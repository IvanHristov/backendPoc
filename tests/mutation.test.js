const gql = require('graphql-tag')
const createTestServer = require('./helper')

const user = { id: 1, first_name: 'ivan', last_name: 'hristov', email: 'ih95@gmail.com' }
const CREATE_CONTACT = gql`
  mutation {
    createContact(input: {first_name: "ivan", last_name: "hr", email: "ih95@gmc"}) {
      id
      first_name
      last_name
      email
    }
  }
`
const UPDATE_CONTACT = gql`
  mutation {
    updateContact(input: {first_name: "ivan"}, filter: {id:4}) {
      id
      first_name
      last_name
      email
    }
  }
`
const DELETE_CONTACT = gql`
  mutation {
    deleteContact(input: {id: 3}) {
      id
      first_name
      last_name
      email
    }
  }
`

describe('mutation', () => {
  test('createContact', async () => {
    const { mutate } = createTestServer({
      models: {
        Contacts: {
          create: jest.fn(input => ({
            id: 1,
            first_name: input.first_name,
            last_name: input.last_name,
            email: input.email
          }))
        }
      }
    })

    const res = await mutate({ query: CREATE_CONTACT })
    expect(res).toMatchSnapshot()
  }),
    test('updateContact', async () => {
      const { mutate } = createTestServer({
        models: {
          Contacts: {
            update: jest.fn(() => [user])
          }
        }
      })

      const res = await mutate({ query: UPDATE_CONTACT })
      expect(res).toMatchSnapshot()
    }),
    test('deleteContact', async () => {
      const { mutate } = createTestServer({
        models: {
          Contacts: {
            delete: jest.fn(() => [user])
          }
        }
      })

      const res = await mutate({ query: DELETE_CONTACT })
      expect(res).toMatchSnapshot()
    });
})
