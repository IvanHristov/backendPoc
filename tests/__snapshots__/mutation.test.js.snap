// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutation createContact 1`] = `
Object {
  "data": Object {
    "createContact": Object {
      "email": "ih95@gmc",
      "first_name": "ivan",
      "id": "1",
      "last_name": "hr",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutation deleteContact 1`] = `
Object {
  "data": Object {
    "deleteContact": Array [
      Object {
        "email": "ih95@gmail.com",
        "first_name": "ivan",
        "id": "1",
        "last_name": "hristov",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutation updateContact 1`] = `
Object {
  "data": Object {
    "updateContact": Array [
      Object {
        "email": "ih95@gmail.com",
        "first_name": "ivan",
        "id": "1",
        "last_name": "hristov",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
